sympy/core/expr.py

```diff
@@ -144,9 +144,25 @@ def __mul__(self, other):
     def __rmul__(self, other):
         return Mul(other, self)
 
+    def __pow__(self, other, mod=None):
+        if mod is None:
+            return self.___pow__(other)
+        try:
+            _self, other, mod = as_int(self), as_int(other), as_int(mod)
+            if other >= 0:
+                return pow(_self, other, mod)
+            else:
+                from sympy.core.numbers import mod_inverse
+                return mod_inverse(pow(_self, -other, mod), mod)
+        except ValueError:
+            power = self.___pow__(other)
+            if power is NotImplemented:
+                return NotImplemented
+            return power%mod
+
     @_sympifyit('other', NotImplemented)
     @call_highest_priority('__rpow__')
-    def __pow__(self, other):
+    def ___pow__(self, other):
```

Comment:
Why did you rename this?

Reply:
@Upabjojr New `__pow__` calls `___pow__`, the original `__pow__`, if mod is None or not all arguments are integers. (See line 149 and 158 above)

Reply:
I've seen this. But why not just keep one function?

Reply:
@Upabjojr It’s because _sympifyit wrapper and call_highest_priority doesn’t support 3-argument function.

Reply:
Perhaps this could be something like `_pow` as it is a private name. (Three underscores feels excessive:)

Reply:
@jksuom Renamed ___pow__ to _pow and reordered functions.